src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -213,7 +225,7 @@ function BaseVideoPlayer({
             const currentDuration = status.durationMillis || videoDuration * 1000;
             const currentPositon = status.positionMillis || 0;
 
-            if (shouldReplayVideo(status, isVideoPlaying, currentDuration, currentPositon)) {
+            if (shouldReplayVideo(status, isVideoPlaying, currentDuration, currentPositon) && !isEnded) {
```

Comment:
NAB: I think it's ok, but why do we need to check !isEnded here? If it's needed, can we make it part of shouldReplayVideo? I'm going to trust all the testing you guys did so you can explain and fix after I merge it.